giskard/llm/client/copilot.py

```diff
@@ -0,0 +1,95 @@
+from typing import Optional, Sequence
+
+from dataclasses import dataclass
+from logging import warning
+
+from ..config import LLMConfigurationError
+from ..errors import LLMImportError
+from .base import ChatMessage
+from .openai import AUTH_ERROR_MESSAGE, OpenAIClient
+
+try:
+    import openai
+except ImportError as err:
+    raise LLMImportError(flavor="talk") from err
+
+
+@dataclass
+class ToolChatMessage(ChatMessage):
+    name: Optional[str] = None
+    tool_call_id: Optional[str] = None
+    tool_calls: Optional[list] = None
+
+
+def _format_message(msg: ChatMessage) -> dict:
+    """Format chat message.
+    Based on a message's role, include related attributes and exclude non-related.
+    Parameters
+    ----------
+    msg : ChatMessage
+        Message to the LLMClient.
+    Returns
+    -------
+    dict
+        A dictionary with attributes related to the role.
+    """
+    fmt_msg = {"role": msg.role, "content": msg.content}
+    if msg.role == "tool":
+        fmt_msg.update({"name": msg.name, "tool_call_id": msg.tool_call_id})
+    if msg.role == "assistant" and msg.tool_calls:
+        fmt_msg.update({"tool_calls": msg.tool_calls})
+    return fmt_msg
+
+
+class GiskardCopilotClient(OpenAIClient):
+    def complete(
+        self,
+        messages: Sequence[ChatMessage],
+        temperature: float = 1.0,
+        max_tokens: Optional[int] = None,
+        caller_id: Optional[str] = None,
+        tools=None,
+        tool_choice=None,
+        seed: Optional[int] = None,
+        format=None,
+    ) -> ChatMessage:
```

Comment:
Typing Issue, should return a ToolChatMessage

Reply:
ToolChatMessage extends ChatMessage